source/dub/dependencyresolver.d

```diff
@@ -81,13 +81,17 @@ class DependencyResolver(CONFIGS, CONFIG) {
 					pidx = *pi;
 					configs = all_configs[*pi];
 				} else {
-					if (basepack == root_base_pack) configs = [root.config];
-					else configs = getAllConfigs(basepack);
-					all_configs ~= configs;
-					package_indices[basepack] = pidx;
-					package_names[pidx] = basepack;
+				  if (basepack == root_base_pack) configs = [root.config];
+				  else if(ch.depType == DependencyType.required)
```

Comment:
Looking at https://github.com/dlang/dub/blob/9a37109faa3a956d625a95a5946314a71da0ffd7/source/dub/dub.d#L1218-L1227, this should be `ch.depType != DependencyType.optional`.